db/migrations/20210614142629_split_out_job_links.js

```diff
@@ -0,0 +1,88 @@
+
+exports.up = function(knex) {
+  // Modify the jobs table to add a jobID
+  return knex.schema
+  .alterTable('jobs', (t) => {
+    t.uuid('jobID');
+    t.unique('jobID');
+  })
+  .then(() => {
+    // Populate the jobID to match the requestID
+    return knex('jobs').update({
+      jobID: knex.raw('??', ['requestId'])
+    });
+  })
+  .then(() => {
+    // Create the job links table
+    return knex.schema.createTable('job_links', (t) => {
+      t.increments('id')
+        .primary();
+
+      t.uuid('jobID')
+        .notNullable()
+        .references('jobID')
+        .inTable('jobs')
+        .onDelete('CASCADE');
+
+      t.string('href', 4096)
+        .notNullable();
+
+      t.string('type', 255);
+
+      t.string('title', 255);
+
+      t.string('rel', 255);
+
+      t.timestamp('temporalStart');
+
+      t.timestamp('temporalEnd');
+
+      t.string('bbox', 255);
+
+      t.timestamp('createdAt')
+        .notNullable();
+
+      t.timestamp('updatedAt')
+        .notNullable();
+
+      t.index(['jobID']);
+    });
+  })
+  .then(async () => {
+    // Populate the job_links table
+    const jobs = await knex.select(['jobID', '_json_links']).from('jobs').forUpdate();
+    jobs.forEach(async (job) => {
+      let links = job._json_links;
+      if (typeof links === 'string') {
+        links = JSON.parse(links);
+      }
+      if (links && links.length > 0) {
+        links.forEach(async (link) => {
+          const { href, type, title, rel, temporal, bbox } = link;
+          const temporalStart = temporal && temporal.start ? new Date(temporal.start) : undefined;
+          const temporalEnd = temporal && temporal.end ? new Date(temporal.end) : undefined;
+          const now = new Date();
+          const createdAt = now;
+          const updatedAt = now;
+          const bboxString = bbox ? bbox.join(',') : undefined;
```

Comment:
Am I correct that we are not currently and this change will not be persisting things like the STAC catalog link and cloud access link?  I think that's the best option, but I can't tell from the code that that's not ever happening, because of the intermixing of those links.

Reply:
Those three links (cloud-access.sh, cloud-access.json, STAC link) are not saved in the database from my testing. But we do have one link representing the S3 bucket that does get stored for every single request. I was thinking that link shouldn't be stored in the DB and should just be presented by the frontend.

```
{ href: "s3://local-staging-bucket/public/harmony/service-example/f437c85c-4908-4ab9-8673-08d06e1606d8/",
title: "Results in AWS S3. Access from AWS us-west-2 with keys from /cloud-access.sh" }
```